### ISSUE TITLE
Should PythonLexer heuristically delegate to PythonConsoleLexer?

### ISSUE THREAD

**Issue by jeanas** [pygments/pygments#2603]
This PR on the repo for the packaging.python.org guide https://github.com/pypa/packaging.python.org/pull/1428 reveals an easy-to-make mistake in Sphinx and similar tools:

```rest
.. code-block:: python

   >>> some Python console session
```

where the lexer should be `pycon` instead.

This looks pretty common actually:

https://grep.app/search?q=%5C.%5C.%20code-block%3A%3A%20python%5Cn%5Cs%2A%3E%3E%3E&regexp=true

https://grep.app/search?q=%60%60%60python%5Cn%3E%3E%3E&regexp=true

I wonder if there's anything that can be done about it and what. It could be improved in Sphinx/mkdocs/other tools, by emitting a warning. But here's something we could do ourselves: if the input to PythonLexer starts with `>>>`, assume it's a console session and let PythonLexer delegate the whole input to PythonConsoleLexer, with `using()`.

I hesitate between calling this nice or hacky. @anteru @birkenfeld What's your opinion?

**Comment by goyalyashpal** [comment]
> This looks pretty common actually:
> 
> https://grep.app/search?q=%5C.%5C.%20code-block%3A%3A%20python%5Cn%5Cs%2A%3E%3E%3E&regexp=true
> 
> https://grep.app/search?q=%60%60%60python%5Cn%3E%3E%3E&regexp=true

* it _looks_ common indeed looking by the match-file count (620, 4013), 
* but i wonder how many different repos there are... 
* as it's reasonable to assume that if someone picks this _wrong_ lexer once in the repo; they will maintain consistency and be wrong throughout that repo;
* so, 1 repo having say 100 python files would get counted 100 times - which massively would skew the results and doesn't frame a balanced picture of the scenario.



-----
yeah, tried manual brute forcing on the side bar, my assumptions were good enough:

[PaddlePaddle/Paddle]: https://grep.app/search?q=%5C.%5C.%20code-block%3A%3A%20python%5Cn%5Cs%2A%3E%3E%3E&regexp=true&filter%5Brepo%5D%5B0%5D=PaddlePaddle/Paddle
[yidao620c/python3-cookbook]: https://grep.app/search?q=%5C.%5C.%20code-block%3A%3A%20python%5Cn%5Cs%2A%3E%3E%3E&regexp=true&filter%5Brepo%5D%5B0%5D=yidao620c/python3-cookbook
[Oneflow-Inc/oneflow]: https://grep.app/search?q=%5C.%5C.%20code-block%3A%3A%20python%5Cn%5Cs%2A%3E%3E%3E&regexp=true&filter%5Brepo%5D%5B0%5D=Oneflow-Inc/oneflow

* [PaddlePaddle/Paddle] : whopping 366
* [yidao620c/python3-cookbook] : 165
* [Oneflow-Inc/oneflow] : 123
* 20-30 each rest all 5: DenisCarriere/​geocoder, dask/​dask, crazyguitar/​pysheeet, tsifrer/​python-twitch-client, pyexcel/​pyexcel

for the second one (without the line break), the skewness ain't that much:

<img src="https://github.com/pygments/pygments/assets/19423063/f8aa17d7-939f-4b3a-81d3-0639084f8a58" width=300px />

**Comment by goyalyashpal** [comment]
-----
> But here's something we could do ourselves: if the input to PythonLexer starts with `>>>`, assume it's a console session and let PythonLexer delegate the whole input to PythonConsoleLexer, with `using()`.


but overall, i'd say that this would only promote the incorrect use. 
* if ppl don't understand that `python` means python source file, 
   and you start catering to those people, 
* then the question would be how far are you willing to go?
* for python itself; there is ipython console with different format, there are repl / interactive console for _literally_ a million different things (u know this very well already); 

will u _spoonfeed_ those as well?

* it is often tempting to provide sugar to the big fishes, 
* but it's a slippery slope thereafter... 
* you can't easily revert the decision lest nobody likes "breaking changes"
* once u make it work, u can't later say very easily, _"ohkay, get ur stuff together, as i ain't doing this now"_

**Comment by sblondon** [comment]
I agree with @goyalyashpal: helping user so they don't fix the wrong definition seems a bad strategy because it adds complexity in `pygments` instead of fixing the mistake. They probably don't detect they use the wrong lexer because PythonLexer still works more or less properly.

Adding a sentence in the documentation of [PythonLexer](https://pygments.org/docs/lexers/#pygments.lexers.python.PythonLexer) would be a better hint for users IMO: 'Use PythonConsoleLexer for python interpreter output' in `PythonLexer` docstring will help them to read the correct class.

@jeanas I can do a PR if you want.

I also sent a PR to fix the wrong lexer for [Paddle](https://github.com/PaddlePaddle/Paddle/pull/69135) documentation. Let's see if the problem is fixed there.

**Comment by sblondon** [comment]
I sent another [issue](https://github.com/bokeh/bokeh/issues/14137) and associated [PR](https://github.com/bokeh/bokeh/pull/14138). It's interesting because it shows the `python` lexer provides a better rendering than the `python-console` one.
